### PR TITLE
JAMES-2365 Covers search filters for address headers with JMAP integr…

### PR DESCRIPTION
**Pull request by aduprat**
…ation tests

### CODE REVIEW

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java [hunk 1]
```diff
@@ -1004,6 +1004,142 @@ public void getMessageListShouldNotFilterMessagesWhenSubjectFilterMatchesSubject
...
+    public void getMessageListShouldNotFilterMessagesWhenFromFilterMatchesFrom() throws Exception {
+        mailboxProbe.createMailbox(MailboxConstants.USER_NAMESPACE, alice, "mailbox");
+        ComposedMessageId message = mailboxProbe.appendMessage(alice, MailboxPath.forUser(alice, "mailbox"),
+            ClassLoader.getSystemResourceAsStream("eml/mailWithRecipients.eml"), new Date(), false, new Flags());
+        await();
+
+        given()
+            .header("Authorization", aliceAccessToken.serialize())
+            .body(String.format("[[\"getMessageList\", {\"filter\":{\"from\":\"from@james.org\"}}, \"#0\"]]"))
+            .when()
```
Comment (chibenwa) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java]: indent

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java [hunk 2]
```diff
@@ -1004,6 +1004,142 @@ public void getMessageListShouldNotFilterMessagesWhenSubjectFilterMatchesSubject
             .body(ARGUMENTS + ".messageIds", contains(message.getMessageId().serialize()));
     }
 
+    @Test
+    public void getMessageListShouldNotFilterMessagesWhenFromFilterMatchesFrom() throws Exception {
```
Comment (mbaechler) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java]: is "filter" meant as "filter out" or "include" ?

**Comment by aduprat** [comment]
```
Failed tests: 
  JPAJamesServerTest.jpaGuiceServerShouldUpdateQuota:88 expected:<...ames.local (STORAGE [12] 50)
AAAB OK GETQUO...> but was:<...ames.local (STORAGE [0] 50)
AAAB OK GETQUO...>
```
test this please

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java [hunk 3]
```diff
@@ -988,24 +988,160 @@ public void getMessageListShouldNotFilterMessagesWhenTextFilterMatchesBody() thr
...
+            .body("[[\"getMessageList\", {\"filter\":{\"from\":\"to@james.org\"}}, \"#0\"]]")
+        .when()
+            .post("/jmap")
+        .then()
+            .statusCode(200)
+            .body(ARGUMENTS + ".messageIds", empty());
+    }
+
+    @Test
+    public void getMessageListShouldIncludeMessagesWhenToFilterMatchesFrom() throws Exception {
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java]: s/From/To

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java [hunk 4]
```diff
@@ -988,24 +988,160 @@ public void getMessageListShouldNotFilterMessagesWhenTextFilterMatchesBody() thr
...
+            .body("[[\"getMessageList\", {\"filter\":{\"to\":\"to@james.org\"}}, \"#0\"]]")
+        .when()
+            .post("/jmap")
+        .then()
+            .statusCode(200)
+            .body(ARGUMENTS + ".messageIds", contains(message.getMessageId().serialize()));
+    }
+
+    @Test
+    public void getMessageListShouldExcludeMessagesWhenToFilterDoesntMatchFrom() throws Exception {
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java]: s/From/To

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java [hunk 5]
```diff
@@ -988,24 +988,160 @@ public void getMessageListShouldNotFilterMessagesWhenTextFilterMatchesBody() thr
...
+        .when()
             .post("/jmap")
-            .then()
+        .then()
+            .statusCode(200)
+            .body(ARGUMENTS + ".messageIds", empty());
+    }
+
+    @Test
+    public void getMessageListShouldIncludeMessagesWhenCcFilterMatchesFrom() throws Exception {
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java]: s/From/Cc

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java [hunk 6]
```diff
@@ -988,24 +988,160 @@ public void getMessageListShouldNotFilterMessagesWhenTextFilterMatchesBody() thr
...
+        .when()
+            .post("/jmap")
+        .then()
             .statusCode(200)
             .body(ARGUMENTS + ".messageIds", contains(message.getMessageId().serialize()));
     }
 
     @Test
-    public void getMessageListShouldFilterMessagesWhenAttachmentFilterDoesntMatches() throws Exception {
+    public void getMessageListShouldExcludeMessagesWhenCcFilterDoesntMatchFrom() throws Exception {
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java]: s/From/Cc

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java [hunk 7]
```diff
@@ -988,24 +988,160 @@ public void getMessageListShouldNotFilterMessagesWhenTextFilterMatchesBody() thr
...
+            .body("[[\"getMessageList\", {\"filter\":{\"cc\":\"bcc@james.org\"}}, \"#0\"]]")
+        .when()
+            .post("/jmap")
+        .then()
+            .statusCode(200)
+            .body(ARGUMENTS + ".messageIds", empty());
+    }
+
+    @Test
+    public void getMessageListShouldIncludeMessagesWhenBccFilterMatchesFrom() throws Exception {
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java]: s/From/Bcc

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java [hunk 8]
```diff
@@ -988,24 +988,160 @@ public void getMessageListShouldNotFilterMessagesWhenTextFilterMatchesBody() thr
...
+            .body("[[\"getMessageList\", {\"filter\":{\"bcc\":\"bcc@james.org\"}}, \"#0\"]]")
+        .when()
+            .post("/jmap")
+        .then()
+            .statusCode(200)
+            .body(ARGUMENTS + ".messageIds", contains(message.getMessageId().serialize()));
+    }
+
+    @Test
+    public void getMessageListShouldExcludeMessagesWhenBccFilterDoesntMatchFrom() throws Exception {
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java]: s/From/Bcc

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java [hunk 9]
```diff
@@ -988,24 +988,160 @@ public void getMessageListShouldNotFilterMessagesWhenTextFilterMatchesBody() thr
...
+            .body("[[\"getMessageList\", {\"filter\":{\"bcc\":\"to@james.org\"}}, \"#0\"]]")
+        .when()
+            .post("/jmap")
+        .then()
+            .statusCode(200)
+            .body(ARGUMENTS + ".messageIds", empty());
+    }
+
+    @Test
+    public void getMessageListShouldExcludeMessagesWhenAttachmentFilterDoesntMatches() throws Exception {
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java]: s/Matches/Match

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**